src/services/userService.ts

```diff
@@ -1,42 +1,45 @@
 import { AxiosInstance } from 'axios';
 import { BaseError, httpStatusCodes } from '../errors/index.js';
 import { Assert, Helper, isErrorRsp } from '../helpers/index.js';
-import {
-  GenericRsp,
-  UserApi,
-  UserCreateReq,
-  UserCreateRsp,
-  UserDeleteReq,
-  UserGetRsp,
-  UserListRsp,
-} from '../generated/index.js';
+import { UserCreateReq, UsersApi, User, UserStatus, GenericRsp } from '../generated/api.js';
 
 export interface UserInterface {
-  create(req: UserCreateReq): Promise<UserCreateRsp>;
-  delete(id: string, req: UserDeleteReq): Promise<GenericRsp>;
-  get(id: string, remoteAddr?: string, userAgent?: string): Promise<UserGetRsp>;
-  list(
-    remoteAddr?: string,
-    userAgent?: string,
-    sort?: string,
-    filter?: string[],
-    page?: number,
-    pageSize?: number,
-  ): Promise<UserListRsp>;
+  create(req: UserCreateReq): Promise<User>;
+  create(fullName: string, status: UserStatus, explicitWebauthnID: string): Promise<User>;
+  createActiveByName(fullName: string): Promise<User>;
+  delete(id: string): Promise<GenericRsp>;
+  get(id: string): Promise<User>;
 }
 
-class User implements UserInterface {
-  private client: UserApi;
+class UserService implements UserInterface {
+  private client: UsersApi;
 
   constructor(axios: AxiosInstance) {
     Assert.notNull(axios, 'User Axios instance must not be null');
-    this.client = new UserApi(undefined, '', axios);
+    this.client = new UsersApi(undefined, '', axios);
   }
 
-  async create(req: UserCreateReq): Promise<UserCreateRsp> {
-    Assert.notNull(req, 'User.create() "req" param must not be null');
+  async create(req: UserCreateReq): Promise<User>;
```

Comment:
interesting solution, please let's talk about this briefly, I would like to understand the benefits of this notation :)